src/services/shell.service.js

```diff
@@ -2,18 +2,49 @@
 /**
  * Thin wrapper around electron `shell`, and other shell-like functions
  */
-import { shell } from 'electron';
+import { shell, remote } from 'electron';
 import launchEditor from 'react-dev-utils/launchEditor';
 import { exec } from 'child_process';
 
 import type { Project } from '../types';
 
+const { BrowserWindow } = remote;
+
+const openedWindows = {};
+
 export const openProjectInFolder = (project: Project) =>
   shell.openItem(project.path);
 
 export const openProjectInEditor = (project: Project) =>
   launchEditor(project.path, 1, 1);
 
+export const openWindow = (url: string) => {
+  const urlKey = url.replace(/(\/|:|\.)/g, '');
+
+  if (openedWindows[urlKey]) {
+    try {
+      return openedWindows[urlKey].show();
+    } catch (err) {
+      // swallow error - Window could be closed
```

Comment:
It might be good to make a note of what the error is, exactly, when the window is closed (maybe see what `err.code` is?), and then to throw a new error if it's not what was expected.

I was bitten at work recently by a try/catch designed to catch a specific error, but it was swallowing a totally different error, and it made debugging really hard

Reply:
Good catch. I'll add a handling. The error I'm swallowing here is `Object has been destroyed`. 
Not sure why I'm getting it because I'm already checking if the object is defined.